### PR TITLE
Modified Table to not inherit from Blob anymore

### PR DESCRIPTION
**Pull request by jessemapel**
I am still running tests on this to ensure that nothing random broke but I want to get it up here for input.

### CODE REVIEW

File: isis/src/base/objs/Cube/Cube.cpp [hunk 1]
```diff
@@ -914,6 +916,12 @@ namespace Isis {
   }
 
 
+  void Cube::write(const Table &table) {
+    Blob tableBlob = table.toBlob();
+    write(tableBlob);
+  }
```
Comment (acpaquette) [inline isis/src/base/objs/Cube/Cube.cpp:964]: Similarly with Austin's PR I don't know if we need a write function for the table object specifically. Just adding `write(table.toBlob())` seems okay to me

File: isis/src/base/objs/Cube/Cube.cpp [hunk 2]
```diff
@@ -1838,6 +1846,13 @@ namespace Isis {
   }
 
 
+  Table Cube::readTable(const QString &name) {
+    Blob tableBlob(name, "Table");
+    read(tableBlob);
```
Comment (acpaquette) [inline isis/src/base/objs/Cube/Cube.cpp]: Should there be a try catch around this read?

**Review by acpaquette — COMMENTED** [review]
Overall looks good! :+1: Few things to talk about

File: isis/src/base/objs/Cube/Cube.cpp (hunk 2 above)
Comment (jessemapel) [reply isis/src/base/objs/Cube/Cube.cpp]: Yeah I should add a try catch to rethrow the Blob error with a better message.

File: isis/src/base/objs/Cube/Cube.cpp (hunk 1 above)
Comment (acpaquette) [reply isis/src/base/objs/Cube/Cube.cpp:964]: We (Jesse, Austin, and I) will be discussing this offline and come to a conclusion on how the writes should be handled

**Review by acpaquette — APPROVED** [review]

**Pull request merged**